engine/apps/api/views/schedule.py

```diff
@@ -282,6 +292,102 @@ def filter_events(self, request, pk):
         }
         return Response(result, status=status.HTTP_200_OK)
 
+    def _resolve_schedule(self, events):
+        """Calculate final schedule shifts considering rotations and overrides."""
+        if not events:
+            return []
+
+        # sort schedule events by (type desc, priority desc, start timestamp asc)
+        events.sort(
+            key=lambda e: (
+                -e["calendar_type"] if e["calendar_type"] else 0,  # overrides: 1, shifts: 0, gaps: None
+                -e["priority_level"] if e["priority_level"] else 0,
+                e["start"],
+            )
+        )
+
+        def _merge_intervals(evs):
+            """Keep track of scheduled intervals."""
+            if not evs:
+                return []
+            intervals = [[e["start"], e["end"]] for e in evs]
+            result = [intervals[0]]
+            for e in intervals[1:]:
+                if result[-1][0] <= e[0] <= result[-1][1]:
```

Comment:
Probably worth introducing variables like `last_interval_start`, `last_interval_end` and `event_start` to make this `result[-1][0] <= e[0] <= result[-1][1]:` become `last_interval_start <= event_start <= last_interval_end`, so it's more clear what it happening.